components/match2/commons/match_group_input_util.lua

```diff
@@ -1076,4 +1077,74 @@ function MatchGroupInputUtil.mergeStandaloneIntoMatch(match, standaloneMatch)
 	return match
 end
 
+
+--- The standard way to process a match input.
+---
+--- The Parser injection must have the following functions:
+--- - extractMaps(match, opponents): table[]
+--- - getBestOf(bestOfInput, maps): integer
+--- - calculateMatchScore(maps): fun(opponentIndex): integer
+---
+--- It may optionally have the following functions:
+--- - removeUnsetMaps(maps): table[]
+--- - getExtraData(match, games, opponents): table
+---
+--- Additionally, the Parser may have the following properties:
+--- - DEFAULT_MODE: string
+--- - DATE_FALLBACKS: string[]
+--- - OPPONENT_CONFIG: table
+---@param match table
+---@param Parser any
+---@return table
+function MatchGroupInputUtil.standardProcessMatch(match, Parser)
+	local finishedInput = match.finished --[[@as string?]]
+	local winnerInput = match.winner --[[@as string?]]
+
+	Table.mergeInto(match, MatchGroupInputUtil.readDate(match.date, Parser.DATE_FALLBACKS))
+
+	local opponents = Array.mapIndexes(function(opponentIndex)
+		return MatchGroupInputUtil.readOpponent(match, opponentIndex, Parser.OPPONENT_CONFIG)
+	end)
+
+	local games = Parser.extractMaps(match, opponents)
+	match.bestof = Parser.getBestOf(match.bestof, games)
+	games = Parser.removeUnsetMaps and Parser.removeUnsetMaps(games) or games
+
+	match.links = MatchGroupInputUtil.getLinks(match)
```

Comment:
what about the h2h link?

Reply:
Leaving it for later PRs to implement as needed